File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.js
function solution(a, b) {
    var answer = 0;
    
    if((a % 2) && (b % 2)) {
        answer = (a*a) + (b*b);
    } else if(!(a % 2) && !(b % 2)) {
        answer = Math.abs(a-b);
    } else {
        answer = 2 * (a+b)
    }
    
    return answer;
}

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.py
def is_odd(num):
    return num % 2

def solution(a, b):
    answer = 0
    
    if is_odd(a) and is_odd(b):
        return a ** 2 + b ** 2
    
    if not is_odd(a) and not is_odd(b):
        return abs(a - b)
    
    return (a + b) * 2